packages/swap/src/utils/__tests__/autoSlippage.test.ts

```diff
@@ -0,0 +1,137 @@
+      destAsset: 'Usdc',
+      egressAmount: 10n,
+      dcaChunks: 1,
+    });
+
+    expect(result).toEqual(0.58);
+  });
+
+  it('should return the correct value for ETH -> USDC for amount >20% of deployed liquidity', async () => {
+    jest.mocked(getUndeployedLiquidity).mockResolvedValue(500n);
+    jest.mocked(getDeployedLiquidity).mockResolvedValue(1000n);
+    jest.mocked(getRequiredBlockConfirmations).mockResolvedValue(8);
+
+    // baseSlippage: 1
+    // depositTimeSlippage: 12s * 8 / 60 * 0.05 = 0.08
+    // undeployedLiquiditySlippage: -0.5
+    // deployedLiquiditySlippage: 0.6
+
+    const result = await calculateRecommendedSlippage({
+      srcAsset: 'Eth',
+      destAsset: 'Usdc',
+      egressAmount: 300n,
+      dcaChunks: 1,
+    });
+
+    expect(result).toEqual(1.18);
+  });
+
+  it('should return the correct value for ETH -> USDC for amount > undeployed liquidity and deployed liquidity', async () => {
+    jest.mocked(getUndeployedLiquidity).mockResolvedValue(500n);
+    jest.mocked(getDeployedLiquidity).mockResolvedValue(1000n);
+    jest.mocked(getRequiredBlockConfirmations).mockResolvedValue(8);
+
+    // baseSlippage: 1
+    // depositTimeSlippage: 12s * 8 / 60 * 0.05 = 0.08
+    // undeployedLiquiditySlippage: 0
+    // deployedLiquiditySlippage: 2.6
+
+    const result = await calculateRecommendedSlippage({
+      srcAsset: 'Eth',
+      destAsset: 'Usdc',
+      egressAmount: 1300n,
+      dcaChunks: 1,
+    });
+
+    expect(result).toEqual(3.68);
+  });
+
+  it('should return the correct value for ETH -> USDC for amount > undeployed liquidity but less than deployed liquidity', async () => {
+    jest.mocked(getUndeployedLiquidity).mockResolvedValue(500n);
+    jest.mocked(getDeployedLiquidity).mockResolvedValue(1000n);
+    jest.mocked(getRequiredBlockConfirmations).mockResolvedValue(8);
+
+    // baseSlippage: 1
+    // depositTimeSlippage: 12s * 8 / 60 * 0.05 = 0.08
+    // undeployedLiquiditySlippage: -0.0625
+    // deployedLiquiditySlippage: 1.6
+
+    const result = await calculateRecommendedSlippage({
+      srcAsset: 'Eth',
+      destAsset: 'Usdc',
+      egressAmount: 800n,
+      dcaChunks: 1,
+    });
+
+    expect(result).toEqual(2.62); // 2.6175 rounded
+  });
+
+  it('should return the correct value for BTC -> ETH when swapping 5 BTC with 6 BTC deployed and 2 BTC undeployed', async () => {
+    jest.mocked(getUndeployedLiquidity).mockResolvedValue(120_000n);
+    jest.mocked(getDeployedLiquidity).mockResolvedValue(360_000n);
+    jest.mocked(getRequiredBlockConfirmations).mockResolvedValue(3);
+
+    // baseSlippage: 1
+    // depositTimeSlippage: 600s * 3 / 60 * 0.1 = 3
+    // undeployedLiquiditySlippage: 0
+    // deployedLiquiditySlippage: 1.6666666666666667
+
+    const result = await calculateRecommendedSlippage({
+      srcAsset: 'Btc',
+      destAsset: 'Eth',
+      intermediateAmount: 3000n,
+      egressAmount: 1n,
+      dcaChunks: 100,
+    });
+
+    expect(result).toEqual(5.67);
```

Comment:
I think this is a usual case (the swap amount and available liquidity) that we've daily. >5% slippage feels not so good here :/
probably we need to revise how formula

Reply:
yeah i expect us to tweak this quite a bit until we are happy. in this case, the `depositTimeSlippage` is too high imo

Reply:
do you think we should hide the `recommendedSlippageTolerancePercent` behind a query param for now to not scare integrators?

Reply:
yeah that's a very good idea!
just put it behind `autoSlippageEnabled` query param